Source/SaveStates/SaveState.cs

```diff
@@ -337,19 +361,96 @@ private IEnumerator LoadStateCoro(bool loadDuped)
             ReflectionHelper.CallMethod(HeroController.instance, "FinishedEnteringScene", true, false);
             ReflectionHelper.CallMethod(GameManager.instance, "UpdateUIStateFromGameState");
 
-            if (data.roomSpecificOptions != "0")
+            if (data.roomSpecificOptions != "0" && data.roomSpecificOptions != null)
             {
                 RoomSpecific.DoRoomSpecific(data.saveScene, data.roomSpecificOptions);
             }
 
-            yield return null;
+            //moving this here seems to work now? no need to toggle canvas
+            //preserve correct hp amount
+            bool isInfiniteHp = DebugMod.infiniteHP;
+            DebugMod.infiniteHP = false;
+            //prevent flower break by taking it, then giving it back
+            PlayerData.instance.hasXunFlower = false;
+            PlayerData.instance.health = data.savedPd.health;
+            HeroController.instance.TakeHealth(1);
+            HeroController.instance.AddHealth(1);
+            PlayerData.instance.hasXunFlower = data.savedPd.hasXunFlower;
+            DebugMod.infiniteHP = isInfiniteHp;
+
+            //this fixes actual lifeblood not just charms, and obsoletes UPDATE BLUE HEALTH
+            int healthBlue = data.savedPd.healthBlue;
+            for (int i = 0; i < healthBlue; i++)
+            {
+                EventRegister.SendEvent("ADD BLUE HEALTH");
+            }
+
+
+            //removes things like bench storage no clip float etc
+            if (DebugMod.settings.SaveStateGlitchFixes) SaveStateGlitchFixes();
 
-            GameCameras.instance.hudCanvas.transform.Find("Health").gameObject.SetActive(false);
-            GameCameras.instance.hudCanvas.transform.Find("Health").gameObject.SetActive(true);
+            //Benchwarp fixes courtesy of homothety, needed since savestates are now performed while paused
+            // Revert pause menu timescale
+            Time.timeScale = 1f;
+            GameManager.instance.FadeSceneIn();
+
+            // We have to set the game non-paused because TogglePauseMenu sucks and UIClosePauseMenu doesn't do it for us.
+            GameManager.instance.isPaused = false;
+
+            // Restore various things normally handled by exiting the pause menu. None of these are necessary afaik
+            GameCameras.instance.ResumeCameraShake();
+            if (HeroController.SilentInstance != null)
```

Comment:
Why silent instance instead of the normal one?

Reply:
It's homothethys code not mine